mm2src/lp_price.rs

```diff
@@ -0,0 +1,336 @@
+                    debug!(
+                        "Unable to fetch tickers price. Tickers ({}/{})",
+                        base_price_infos.last_price, rel_price_infos.last_price
+                    );
+                    return None;
+                }
+
+                rate_infos.base_provider = base_price_infos.price_provider.clone();
+                rate_infos.rel_provider = rel_price_infos.price_provider.clone();
+                rate_infos.last_updated_timestamp =
+                    if base_price_infos.last_updated_timestamp <= rel_price_infos.last_updated_timestamp {
+                        Some(base_price_infos.last_updated_timestamp)
+                    } else {
+                        Some(rel_price_infos.last_updated_timestamp)
+                    };
+                rate_infos.base_price = base_price_infos.last_price.clone();
+                rate_infos.rel_price = rel_price_infos.last_price.clone();
+                rate_infos.price = &base_price_infos.last_price / &rel_price_infos.last_price;
+                Some(rate_infos)
+            },
+            None => None,
+        }
+    }
+}
+
+pub async fn process_price_request(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    debug!("Fetching price from: {}", price_url);
+    let (status, headers, body) = slurp_url(price_url).await?;
+    let (status_code, body, _) = (status, std::str::from_utf8(&body)?.trim().into(), headers);
+    if status_code != StatusCode::OK {
+        return MmError::err(PriceServiceRequestError::HttpProcessError(body));
+    }
+    let model: HashMap<String, TickerInfos> = serde_json::from_str(&body)?;
+    Ok(TickerInfosRegistry(model))
+}
+
+pub async fn fetch_price_tickers(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    let model = process_price_request(price_url).await?;
+    debug!("price registry size: {}", model.0.len());
+    Ok(model)
+}
+
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct CEXRates {
+    pub base: BigDecimal,
+    pub rel: BigDecimal,
+}
+
+// Fetcher funtion to test fetching latest prices from a endoint.
+async fn try_price_fetcher_endpoint(
+    endpoint: &str,
+    base: &str,
+    rel: &str,
+) -> Result<CEXRates, MmError<PriceServiceRequestError>> {
+    match process_price_request(endpoint).await {
+        Ok(response) => match response.get_cex_rates(base.to_string(), rel.to_string()) {
+            Some(fiat_price) => {
+                let (base_usd_price, rel_usd_price) = fiat_price.get_rate_price();
+                Ok(CEXRates {
+                    base: base_usd_price,
+                    rel: rel_usd_price,
+                })
+            },
+            None => {
+                let cex: Option<CEXRates> = None;
+                cex.or_mm_err(|| PriceServiceRequestError::Internal("Couldn't fetch price".to_string()))
+            },
+        },
+        Err(e) => MmError::err(e.into_inner()),
+    }
+}
```

Comment:
I left these comments to give you ideas how you can make the code a bit shorter and easier to read
https://github.com/KomodoPlatform/atomicDEX-API/pull/1282#discussion_r869858694
https://github.com/KomodoPlatform/atomicDEX-API/pull/1282#discussion_r869861607

What I meant:
```rust
// Fetcher funtion to test fetching latest prices from a endoint.
async fn try_price_fetcher_endpoint(
    endpoint: &str,
    base: &str,
    rel: &str,
) -> Result<CEXRates, MmError<PriceServiceRequestError>> {
    let response = process_price_request(endpoint).await?;
    let fiat_price = response
        .get_cex_rates(base.to_string(), rel.to_string())
        .or_mm_err(|| PriceServiceRequestError::Internal("Couldn't fetch price".to_string()))?;
    let (base_usd_price, rel_usd_price) = fiat_price.get_rate_price();
    Ok(CEXRates {
        base: base_usd_price,
        rel: rel_usd_price,
    })
}
```

Reply:
done